packages/utils/telemetry-utils/src/telemetryEventBatcher.ts

```diff
@@ -71,34 +53,13 @@ export class TelemetryEventBatcher<TMetrics extends string> {
 		private readonly threshold: number,
 	) {}
 
-	/**
-	 * Executes the specified code, keeping statistics of its execution time and the telemetry properties it returns, and when the {@link TelemetryEventBatcher.threshold} is reached it logs a performance event which includes the maxes and averages.
-	 * @param codeToMeasure - The code to be executed and measured.
-	 * @param customData - Custom data to be logged.
-	 *
-	 * @returns Whatever the passed-in code block returns.
-	 */
-	public measure<T extends IMeasuredCodeResult<TMetrics>>(codeToMeasure: () => T): T {
-		const start = performance.now();
-		const returnValue = codeToMeasure();
-		const duration = performance.now() - start;
-
-		this.accumulatedDuration += duration;
-
-		if (returnValue.telemetryProperties) {
-			this.accumulateAndLog(returnValue.telemetryProperties);
-		}
-
-		return returnValue;
-	}
-
 	/**
 	 * Accumulates the custom data and sends it to the logger every {@link TelemetryEventBatcher.threshold} calls.
 	 *
 	 * @param customData -
 	 * A record storing the custom data to be accumulated and eventually logged.
 	 */
-	private accumulateAndLog(customData: Record<TMetrics, number>): void {
+	public accumulateAndLog(customData: Record<TMetrics, number>): void {
```

Comment:
One small quirk of the event logged by this class is that it won't have a `duration` property, but `avgDuration` and `maxDuration`. It's unconventional but fine since `duration` is optional in the performance event interface, but maybe something we want to come back to at some point.